### PR TITLE
Remove SMHasher lib and include MurmurHash2 in c_util.

### PR DESCRIPTION
**Pull request by darobs**
@anporumb 's suggestion, only include the files we are actually using directly into c_util.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -235,10 +221,14 @@ SOURCE_GROUP(devdoc FILES ${c_util_md_files})
 #this is the product (a library)
 add_library(c_util ${c_util_c_files} ${c_util_cpp_files} ${c_util_h_files} ${c_util_md_files})
 
-target_link_libraries(c_util c_pal SMHasherSupport)
+target_link_libraries(c_util c_pal)
 
 target_include_directories(c_util PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/inc>)
 
+if (EXISTS ${CMAKE_CURRENT_LIST_DIR}/deps/smhasher/src/MurmurHash2.cpp)
+    target_include_directories(c_util PRIVATE ${CMAKE_CURRENT_LIST_DIR}/deps/smhasher/src)
```
Comment (anporumb) [inline CMakeLists.txt:229]: >PRIVATE [](http://example.com/codeflow?start=38&length=7)

PUBLIC

File: CMakeLists.txt [hunk 2]
```diff
@@ -213,6 +194,11 @@ set(c_util_h_files
     ./inc/c_util/async_op.h
 )
 
+if (EXISTS ${CMAKE_CURRENT_LIST_DIR}/deps/smhasher/src/MurmurHash2.cpp)
+    set(c_util_cpp_files ${c_util_cpp_files} ${CMAKE_CURRENT_LIST_DIR}/deps/smhasher/src/MurmurHash2.cpp)
+    set(c_util_h_files ${c_util_h_files} ${CMAKE_CURRENT_LIST_DIR}/deps/smhasher/src/MurmurHash2.h)
+endif()
```
Comment (anporumb) [inline CMakeLists.txt:200]: >endif [](http://example.com/codeflow?start=0&length=5)

shouldn't here be some compilation exclusions with warnings disabled?

File: CMakeLists.txt [hunk 3]
```diff
@@ -235,10 +221,14 @@ SOURCE_GROUP(devdoc FILES ${c_util_md_files})
 #this is the product (a library)
 add_library(c_util ${c_util_c_files} ${c_util_cpp_files} ${c_util_h_files} ${c_util_md_files})
 
-target_link_libraries(c_util c_pal SMHasherSupport)
+target_link_libraries(c_util c_pal)
 
 target_include_directories(c_util PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/inc>)
 
+if (EXISTS ${CMAKE_CURRENT_LIST_DIR}/deps/smhasher/src/MurmurHash2.cpp)
```
Comment (dcristoloveanu) [inline CMakeLists.txt:228]: >(EXISTS ${CMAKE_CURRENT_LIST_DIR}/deps/smhasher/src/MurmurHash2.cpp) [](http://example.com/codeflow?start=3&length=68)

Consider having a function that encapsulates this path.
Like smhasher_exists

File: tests/reals/CMakeLists.txt [hunk 4]
```diff
@@ -43,6 +43,11 @@ set(c_util_reals_h_files
     real_uuid_string_renames.h
 )
 
+if (EXISTS ${CMAKE_CURRENT_LIST_DIR}/../../deps/smhasher/src/MurmurHash2.cpp)
+    set(c_util_reals_cpp_files ${c_util_reals_cpp_files} ${CMAKE_CURRENT_LIST_DIR}/../../deps/smhasher/src/MurmurHash2.cpp)
+    set(c_util_reals_h_files ${c_util_reals_h_files} ${CMAKE_CURRENT_LIST_DIR}/../../deps/smhasher/src/MurmurHash2.h)
```
Comment (anporumb) [inline tests/reals/CMakeLists.txt:48]: talking about "reals" the c_util and c_util_reals will end up exporting the same symbol from murmurhash, so inside c_utils_reals that needs to be renamed.

Parth's tool that checks that will trigger.

File: CMakeLists.txt (hunk 3 above)
Comment (anporumb) [inline CMakeLists.txt:228]: when is the case that this file does not exist?

**Comment by anporumb** [comment]
> #Copyright (c) Microsoft. All rights reserved.

please add here (and everywhere in this c_util)

"#Licensed under the MIT license. See LICENSE file in the project root for full license information."

---
Refers to: tests/hash_ut/CMakeLists.txt:1 in a935c65. [](commit_id = a935c65bcc40df8c21ae81d000be38c7d7484242, deletion_comment = False)

**Review by anporumb — APPROVED** [review]
:shipit:

**Review by dcristoloveanu — APPROVED** [review]

**Pull request merged**